src/main/java/org/miracum/streams/fhirtoserver/SendToServerProcessor.java

```diff
@@ -95,7 +95,7 @@ public SendToServerProcessor(
           public <T, E extends Throwable> void onError(
               RetryContext context, RetryCallback<T, E> callback, Throwable throwable) {
             LOG.warn(
-                "Trying to sent resource to FHIR server caused error. {} attempt.",
+                "Trying to sent resource to FHIR server caused error. {} attempt. Error: {}",
```

Comment:
Throwable/exception as last Parameter already printed the exception including stacktrace, so it is preferable over formatting it as "{}". The problem is, that the exception message isn't particularly helpful. Instead, we should match on the exception type and if it's a fhirexception such as https://hapifhir.io/hapi-fhir/apidocs/hapi-fhir-base/ca/uhn/fhir/rest/server/exceptions/UnprocessableEntityException.html we should retrieve it's Operation outcome and either pretty print it as Json in it's entirety or manually list and log all issues returned by the server.